### ISSUE TITLE
Debugger entered--Lisp error: (void-variable polymode-run-these-before-change-functions-in-other-buffers)

### ISSUE THREAD

**Issue by malcook** [polymode/polymode#309]
I'm getting this error:

```
Debugger entered--Lisp error: (void-variable polymode-run-these-before-change-functions-in-other-buffers)
  (append '(org-before-change-function org-element--cache-before-change org-table-remove-rectangle-highlight) polymode-run-these-before-change-functions-in-other-buffers)
  (set (make-local-variable 'polymode-run-these-before-change-functions-in-other-buffers) (append '(org-before-change-function org-element--cache-before-change org-table-remove-rectangle-highlight) polymode-run-these-before-change-functions-in-other-buffers))
  (let ((last-message (current-message)) (state (cond ((numberp arg) (> arg 0)) (arg t) ((not poly-org-mode))))) (setq poly-org-mode state) (if state (if (buffer-base-buffer) nil (if poly-org-mode (progn (let ((obj ...)) (pm-initialize obj)) (setq poly-org-mode t)))) (let ((base (pm-base-buffer))) (pm-turn-polymode-off t) (switch-to-buffer base))) (set (make-local-variable 'org-src-fontify-natively) nil) (set (make-local-variable 'polymode-move-these-minor-modes-from-old-buffer) (append '(org-indent-mode) polymode-move-these-minor-modes-from-old-buffer)) (set (make-local-variable 'polymode-run-these-before-change-functions-in-other-buffers) (append '(org-before-change-function org-element--cache-before-change org-table-remove-rectangle-highlight) polymode-run-these-before-change-functions-in-other-buffers)) (set (make-local-variable 'polymode-run-these-after-change-functions-in-other-buffers) (append '(org-element--cache-after-change) polymode-run-these-after-change-functions-in-other-buffers)) (if state (progn (pm--run-derived-mode-hooks))) (if (buffer-base-buffer) nil (if (and (called-interactively-p 'any) (or (null (current-message)) (not (equal last-message (current-message))))) (progn (message "org polymode %s" (if state "enabled" "disabled"))))) (force-mode-line-update))
  poly-org-mode()
  set-auto-mode-0(poly-org-mode nil)
  set-auto-mode--apply-alist((("\\.sparql$" . sparql-mode) ("\\.py\\'" . python-mode) ("\\.js$" . js2-mode) ("\\.lua$" . lua-mode) ("\\.yml$" . yaml-mode) ("\\.\\(sql\\)" . sql-mode) ("Guardfile$" . ruby-mode) ("Capfile$" . ruby-mode) ("Gemfile$" . ruby-mode) ("Thorfile$" . ruby-mode) ("[rR]akefile$" . ruby-mode) ("\\.rb$" . ruby-mode) ("\\.ru$" . ruby-mode) ("\\.gemspec$" . ruby-mode) ("\\.rake$" . ruby-mode) ("\\.Rmd$" . poly-markdown+r-mode) ("\\.Rmd" . markdown-mode) ("\\.odc\\'" . archive-mode) ("\\.odf\\'" . archive-mode) ("\\.odi\\'" . archive-mode) ("\\.otp\\'" . archive-mode) ("\\.odp\\'" . archive-mode) ("\\.otg\\'" . archive-mode) ("\\.odg\\'" . archive-mode) ("\\.ots\\'" . archive-mode) ("\\.ods\\'" . archive-mode) ("\\.odm\\'" . archive-mode) ("\\.ott\\'" . archive-mode) ("\\.odt\\'" . archive-mode) ("\\.[pP][dD][fF]\\'" . pdf-view-mode) ("\\.markdown\\'" . markdown-mode) ("\\.md\\'" . markdown-mode) ("README\\.md\\'" . gfm-mode) ("\\.tsv\\'" . tsv-mode) ("\\.[Cc][Ss][Vv]\\'" . csv-mode) ("\\(?:build\\|profile\\)\\.boot\\'" . clojure-mode) ("\\.cljs\\'" . clojurescript-mode) ("\\.cljc\\'" . clojurec-mode) ("\\.\\(clj\\|dtm\\|edn\\)\\'" . clojure-mode) ("\\.[Ss][Aa][Ss]\\'" . SAS-mode) ("\\.Sout" . S-transcript-mode) ("\\.[Ss]t\\'" . S-transcript-mode) ("\\.Rd\\'" . Rd-mode) ("DESCRIPTION\\'" . conf-colon-mode) ("/Makevars\\(\\.win\\)?\\'" . makefile-mode) ("\\.[Rr]out" . ess-r-transcript-mode) ("CITATION\\'" . ess-r-mode) ("NAMESPACE\\'" . ess-r-mode) ("\\.[rR]profile\\'" . ess-r-mode) ("\\.[rR]\\'" . ess-r-mode) ...) nil nil)
  set-auto-mode()
  normal-mode(t)
  after-find-file(nil t)
...
```

This is kind of bleeding edge, so perhaps I'm asking for trouble.  I'm using 

GNU Emacs 29.0.50 (build 1, x86_64-pc-linux-gnu, X toolkit, cairo version 1.15.12, Xaw3d scroll bars) of 2022-01-19
Org mode version 9.5.2 (release_9.5.2-312-g5d05f5 @ /home/mec/.emacs.d/org-mode/lisp/)
from elpa:
 poly-R-20210930.1921
 poly-markdown-20220117.2351
 poly-noweb-20200316.1315
 poly-org-20220120.1017
 polymode-20220120.1016

and the org file in question is remote over tramp ssh, if that might make a difference

Can I provide any additional diagnostics?

**Comment by vspinu** [comment]
Please update your polymode as well. I haven't linked the versions as yet.

**Comment by malcook** [comment]
>Please update your polymode as well. I haven't linked the versions as yet.

hmm. Isn't polymode-20220120.1016 from elpa the latest?  

It is what I see when I do package-list-packages even after refreshing them.

Should I do something other?

**Comment by vspinu** [comment]
Strange. The variable is [defined](https://github.com/polymode/polymode/blob/61e2326731cad373432126e7b12a153bed7d1e89/polymode-core.el#L1399) in the core. 
Maybe your polymode is loaded from a different location by accident? Can you navigate to a symbol and check? Also check if this variable is indeed defined in your version of polymode-core.el?

**Comment by malcook** [comment]
It seems my running emacs --daemon environment referred still to previously installed version of polymode.

Simply restarting my --daemon and reconnecting solved my issue, and polymode-20220120.1016 is serving me well again.

Undoubtedly "operator error".

Thanks for you attention.

FWIW: this happy poly-org user mixes & matches R, bash, sql, perl as needed.  Thx for polymode!
